parsl/app/errors.py

```diff
@@ -128,10 +133,19 @@ def reraise(self) -> None:
         # specific exception type.
         logger.debug("Reraising exception of type {}".format(t))
 
-        v = dill.loads(self.e_value)
-        tb = self.e_traceback.as_traceback()
+        v = self.get_exception()
+
+        reraise(t, v, v.__traceback__)
 
-        reraise(t, v, tb)
+    def get_exception(self) -> Exception:
```

Comment:
the return value of this can be a BaseException, I think? (because __init__ takes a BaseException, so e_value would contain a serialised BaseException).

This doesn't cause a type error here because dill.dumps/loads doesn't preserve types, but in some other work I've done that causes a problem.

Reply:
Yes, you are right, BaseException would have been correct.